php/class-post-type.php

```diff
@@ -675,4 +685,140 @@ public function hide_disabled_publishing_actions( $post ) {
+		return $bulk_actions;
+	}
+
+	/**
+	 * Handle bulk actions.
+	 *
+	 * @param string $redirect_to url to redirect to.
+	 * @param string $do_action   current action.
+	 * @param array  $post_ids    post ids.
+	 *
+	 * @return string url.
+	 */
+	public function handle_snapshot_bulk_actions( $redirect_to, $do_action, $post_ids ) {
+		if ( 'merge_snapshot' !== $do_action ) {
+			return $redirect_to;
+		}
+		$posts = array_map( 'get_post', $post_ids );
+		if ( count( $posts ) <= 1 ) {
+			return empty( $redirect_to ) ? add_query_arg( array( 'merge-error' => 1 ) ) : add_query_arg( array( 'merge-error' => 1 ), $redirect_to );
+		}
+
+		$allowed_status = array( 'draft', 'future' );
+		$posts = array_filter( $posts, function( $p ) use ( $allowed_status ) {
+			return in_array( $p->post_status, $allowed_status, true );
+		} );
+
+		if ( count( $posts ) <= 1 ) {
+			return empty( $redirect_to ) ? add_query_arg( array( 'merge-error' => 2 ) ) : add_query_arg( array( 'merge-error' => 2 ), $redirect_to );
+		}
+
+		usort( $posts, function( $a, $b ) {
+			$compare_a = $a->post_modified;
+			$compare_b = $b->post_modified;
+			if ( '0000-00-00 00:00:00' === $compare_a ) {
+				$compare_a = $a->post_date;
+			}
+			if ( '0000-00-00 00:00:00' === $compare_b ) {
+				$compare_a = $b->post_date;
+			}
+			return strtotime( $compare_a ) - strtotime( $compare_b );
+		} );
+
+		foreach ( $posts as $post ) {
+			$post->post_content = $this->get_post_content( $post );
+		}
+		$content = wp_list_pluck( $posts, 'post_content' );
+		$conflict_keys = call_user_func_array( 'array_intersect_key', $content );
+		$merge_value = call_user_func_array( 'array_merge', $content );
+
+		foreach ( $conflict_keys as $key => $conflict_val ) {
+			$original_values = array();
+			foreach ( $posts as $post ) {
+				if ( isset( $post->post_content[ $key ] ) ) {
+					$original_values[] = $post->post_content[ $key ];
+				}
+			}
+			array_pop( $original_values );
+			$merge_value[ $key ]['merge_conflict'] = $original_values;
+		}
+		$post_id = $this->save( array(
+			'uuid' => Customize_Snapshot_Manager::generate_uuid(),
+			'status' => 'draft',
+			'data' => $merge_value,
+			'post_date' => current_time( 'mysql', false ),
+			'post_date_gmt' => current_time( 'mysql', true ),
+		) );
+		$redirect_to = get_edit_post_link( $post_id, 'raw' );
+		return $redirect_to;
+	}
+
+	/**
+	 * Insert script for adding merge snapshot bulk action polyfill.
+	 */
+	public function snapshot_merge_print_script() {
+		global $post_type;
+		if ( self::SLUG === $post_type ) {
+			?>
+			<script type="text/javascript">
+		      jQuery(document).ready(function() {
+			      jQuery('<option>').val('merge_snapshot').text('<?php esc_html_e( 'Merge Snapshot', 'customize-snapshots' ); ?>').appendTo("select[name='action']");
+			      jQuery('<option>').val('merge_snapshot').text('<?php esc_html_e( 'Merge Snapshot', 'customize-snapshots' ); ?>').appendTo("select[name='action2']");
```

Comment:
These lines aren't right. There could be a translation string that has an apostrophe and this would cause a JS error. I suggest refactoring as follows:

```js
jQuery( function( $ ) {
    var optionText = <?php echo wp_json_encode( __( 'Merge Snapshot', 'customize-snapshots' ) ); ?>;
    $( 'select[name="action"], select[name="action2"]' ).each( function() {
        var option = $( '<option>', {
            text: optionText,
            value: 'merge_snapshot'
        } );
        $( this ).append( option );
    } );
} );
```